### PR TITLE
Be careful for ## in strings with Ruby

### PR DESCRIPTION
**Pull request by pschambacher**
:octopus::computer:

/cc @zendesk/octo 
### Description

Be more careful about double-comments in strings
This could be a fix for [this issue](https://github.com/zendesk/zendesk/pull/21339)
### Steps to merge
- [ ] :+1: from the team
### Risks
- Low: This won't catch any documentation comment directly appended to the code

``` ruby
my.line.of(:code)## This would not be extracted anymore
```

``` javascript
my().line().of('code')/// Neither would this
```

### CODE REVIEW

**Comment by grosser** [comment]
:heart: :+1:

File: lib/double_doc/doc_extractor.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 module DoubleDoc
   class DocExtractor
     TYPES = {
-      'rb' => /\s*##\s?(.*)$/,
-      'js' => %r{\s*///\s?(.*)$}
+      'rb' => /(^\s*|\s+)##\s?(?<documentation_line>.*)$/,
+      'js' => %r{(^\s*|\s+)///\s?(?<documentation_line>.*)$}
```
Comment (steved) [inline lib/double_doc/doc_extractor.rb]: `(^|\s+)` is probably good enough

Comment (steved) [inline lib/double_doc/doc_extractor.rb]: Or maybe even `^\s*`?

Comment (pschambacher) [inline lib/double_doc/doc_extractor.rb]: Second one would be good if we don't allow inline documentation. Do we?

Comment (steved) [inline lib/double_doc/doc_extractor.rb]: That's the real question, too many repos to check. You'd want to compare the generated documentation for them. I think your change is the safest.

**Comment by gardhauge** [comment]
@pschambacher can you remember why this never got merged? This issue is still live and kicking.

**Comment by pschambacher** [comment]
@gardhauge probably because Travis is red. I'll need to rebase and get this updated.

**Comment by pschambacher** [comment]
@grosser @gardhauge @sandlerr Can I has another review?

File: test/doc_extractor_test.rb [hunk 2]
```diff
@@ -6,6 +6,7 @@ def self.it_acts_like_an_extractor
     it "extracts documentation" do
       subject.must_match(/this line should be extracted/)
       subject.must_match(/this line should also be extracted/)
+      subject.must_match(/but this line should/)
```
Comment (grosser) [inline test/doc_extractor_test.rb]: prefer `.must_include` for nicer error messages

**Comment by grosser** [comment]
👍

**Comment by pschambacher** [comment]
I don't have the permission to release my tag

**Comment by grosser** [comment]
@staugaard is the owner

**Comment by staugaard** [comment]
Who wan't to be added as owners? Please email me your rubygems.org email addresses.

**Comment by grosser** [comment]
sent

**Comment by gardhauge** [comment]
Another thing, to the new owners I guess: rubygems.org has http://staugaard.github.io/double_doc set as the homepage for double_doc, it gives a 404. Probably should just be https://github.com/zendesk/double_doc/ ?
@grosser @pschambacher

**Comment by grosser** [comment]
Yeah, just makes the code harder to find ... real addres would be
http://zendesk.github.io/double_doc

**Comment by grosser** [comment]
2.1.1 is out now

**Pull request merged**